ui/src/helpers/api.js

```diff
@@ -48,11 +49,27 @@ export default class Api {
     }
 
     validateQuery = async (query) => {
+        if (this.source) {
+            this.source.cancel();
+        }
+        this.source = CancelToken.source();
+
         const form = new FormData();
         form.append('query', query)
         const response = await this.client.post(`/query/validate`, form, {
-            cancelToken: source.token
+            cancelToken: this.source.token
+        }).catch(function (thrown) {
+            if (axios.isCancel(thrown)) {
+                console.log('Validate canceled');
```

Comment:
I think this can be removed (it is probably good for debugging but not when we are going to merge)